### PR TITLE
At least fixed one freeze on dialog exit

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -1625,8 +1625,9 @@ void Emulator::Stop(bool restart)
 	cpu_thread::stop_all();
 	g_fxo->reset();
 
-	while (thread_ctrl::get_count())
+	while (u32 x = thread_ctrl::get_count())
 	{
+		sys_log.fatal("Waiting for %u threads...");
 		std::this_thread::sleep_for(10ms);
 	}
```
Comment (sampletext32) [inline rpcs3/Emu/System.cpp]: There I think should be X passed to log.fatal.

Comment (Nekotekina) [reply rpcs3/Emu/System.cpp]: Thanks, also increased the timeout.

File: rpcs3/Emu/RSX/Overlays/overlay_message_dialog.cpp [hunk 2]
```diff
@@ -256,12 +256,18 @@ namespace rsx
 						}
 						else
 						{
-							while (!exit)
+							while (!exit && thread_ctrl::state() == thread_state::created)
 							{
 								refresh();
 
 								// Only update the screen at about 60fps since updating it everytime slows down the process
 								std::this_thread::sleep_for(16ms);
```
Comment (elad335) [inline rpcs3/Emu/RSX/Overlays/overlay_message_dialog.cpp]: maybe use thread_ctrl::wait() here.

Comment (Nekotekina) [reply rpcs3/Emu/RSX/Overlays/overlay_message_dialog.cpp]: Well, I'm not touching this parts of code.

**Comment by Nekotekina** [comment]
Pushing because need fixes

**Pull request merged**